### PR TITLE
Expanding ruff application of docstring rules

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

I added blanket coverage to all `D2` and `D3` docstring rules in `ruff`.  

ALMOST all of these were fully automatable via `ruff --fix .`. 

## Why is the change being made?

This is part of the on-going work to get the ARMI `ruff.toml` up-to-date. But I am also doing it one (or two) rules at a time, so my reviewers can actually review all these changes.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/apps.py [hunk 1]
```diff
@@ -231,7 +229,7 @@ def getParamRenames(self) -> Dict[str, str]:
         return renames
 
     def registerUserPlugins(self, pluginPaths):
-        """
+        r"""
```
Comment (opotowsky) [inline armi/apps.py:232]: @john-science why is the `r` being added? I'm assuming the docs built fine before without it?

File: armi/materials/uraniumOxide.py [hunk 2]
```diff
@@ -131,9 +131,7 @@ def applyInputParams(
         material.FuelMaterial.applyInputParams(self, *args, **kwargs)
 
     def setDefaultMassFracs(self) -> None:
-        r"""
-        UO2 mass fractions. Using Natural Uranium without U234.
-        """
+        r"""UO2 mass fractions. Using Natural Uranium without U234."""
```
Comment (opotowsky) [inline armi/materials/uraniumOxide.py:134]: Interesting that `ruff` adds `r` when needed but doesn't remove it when it's not.

Not asking you to comb through these, just making an observation :-)

**Review by opotowsky — APPROVED** [review]
Looks good to me!

File: armi/materials/uraniumOxide.py (hunk 2 above)
Comment (john-science) [reply armi/materials/uraniumOxide.py:134]: So, I BELIEVE there is a different `ruff` rule for removing the unnecessary r's.  But that's just not part of this PR.

Soon!

Comment (opotowsky) [reply armi/materials/uraniumOxide.py:134]: <3

**Pull request merged**